edx_when/api.py

```diff
@@ -398,11 +398,29 @@ def get_schedules_with_due_date(course_id, assignment_date):
     # Using those relative dates, get all Schedules that have a "hit" by working backwards to the start_date
     rel_start_dates = [assignment_date - rel_date for rel_date in rel_dates]
 
-    # Exclude any user that has an overridden date for a course on the specified day so there aren't duplicates
-    return Schedule.objects.filter(
-        enrollment__course_id=course_id,
-        start_date__date__in=rel_start_dates,
-    ).exclude(enrollment__user_id__in=user_ids).select_related('enrollment') | user_date_overridden_schedules
+    if rel_start_dates:
+        # Exclude any user that has an overridden date for a course on the specified day so there aren't duplicates
+        schedules = Schedule.objects.filter(
+            enrollment__course_id=course_id,
+            enrollment__is_active=True,
+            start_date__date__in=rel_start_dates,
+        ).exclude(enrollment__user_id__in=user_ids).select_related('enrollment') | schedules
+
+    # Add in all users with relative dates to exclude from the absolute dates query to prevent duplicates
+    user_ids = schedules.all().values_list('enrollment__user_id', flat=True).distinct()
```

Comment:
Do you maybe want to exclude by enrollment id not user id?

Oh, I see, this is all in the context of a single course, so there won't be multiple enrollments per user. I guess it's still an extra join table. But not worth fixing.

Reply:
I thought about that but decided it's easier to avoid needing to find the enrollments from the above section for UserDates, figured it was better to save the one query and just do the join, but maybe it's not? Any ideas?